karton/core/task.py

```diff
@@ -103,12 +124,32 @@ def __init__(
         self.headers = {**headers, **headers_persistent}
         self._headers_persistent_keys = set(headers_persistent.keys())
         self.status = TaskState.DECLARED
+        self.headers = headers
+        self.status = _status or TaskState.DECLARED
 
-        self.last_update: float = time.time()
+        self.last_update: float = _last_update or time.time()
         self.priority = priority or TaskPriority.NORMAL
 
         self.payload = dict(payload)
         self.payload_persistent = dict(payload_persistent)
+        # This field should be set to different value only during deserialization
+        # For tasks created by current code, it should be set to latest
+        self.revision = self.REVISION
+
+    @property
+    def fquid(self) -> str:
+        """
+        Fully-qualified task uid in form <root_uid>:<task_uid>.
+
+        Used as a primary task identifier for storage in Redis since Karton 5.1.
+        """
+        if self.REVISION == 1:
```

Comment:
```suggestion
        if self.revision == 1:
```

We want to check the object attribute, not the class one